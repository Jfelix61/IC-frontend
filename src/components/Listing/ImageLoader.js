import React, { useState } from "react";
import PropTypes from "prop-types";

//Wrapper for image; changes class of image to loaded after lazy loading
const ImageLoader = ({ src, alt, className }) => {
  const [classNames, setClassNames] = useState(
    `${className ? className : ""} img-loading`
  );
  const onLoad = () => {
    setClassNames(`${className ? className : ""} img-loaded`);
  };

  return <img src={src} alt={alt} onLoad={onLoad} className={classNames} />;
};

ImageLoader.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  className: PropTypes.string
};

export default ImageLoader;
