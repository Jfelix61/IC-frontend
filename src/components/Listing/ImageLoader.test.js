import React from "react";
import { render } from "react-testing-library";

import ImageLoader from "./ImageLoader";

describe("ImageLoader", () => {
  it("renders without crashing", () => {
    const { asFragment } = render(
      <ImageLoader
        src="https://via.placeholder.com/150"
        alt="image"
        className="image"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
