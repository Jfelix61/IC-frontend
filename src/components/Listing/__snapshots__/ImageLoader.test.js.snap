// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLoader renders without crashing 1`] = `
<DocumentFragment>
  <img
    alt="image"
    class="image img-loading"
    src="https://via.placeholder.com/150"
  />
</DocumentFragment>
`;
